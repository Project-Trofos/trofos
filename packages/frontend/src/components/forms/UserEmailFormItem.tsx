import React from 'react'
import { Form, Input } from 'antd'

type UserEmailFormItemProps = {
    isDisabled?: boolean;
    isRequired?: boolean;
}

export default function UserEmailFormItem({ isDisabled, isRequired } : UserEmailFormItemProps) : JSX.Element {
    return (
        <Form.Item
            label="User Email"
            name="userEmail"
            rules={isRequired ? [{required: true, message: "Please input the user's email!"}] : []}
        >
            <Input disabled={isDisabled}/>
        </Form.Item>
    )
}

UserEmailFormItem.defaultProps = {
    isDisabled: false,
    isRequired: false,
}