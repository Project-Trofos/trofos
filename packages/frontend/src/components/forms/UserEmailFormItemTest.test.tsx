import { render, screen } from '@testing-library/react';
import React from 'react';
import UserEmailFormItem from './UserEmailFormItem';
import FormWrapper from './__testutils__/FormRenderHelper';

describe('UserEmailFormItem test', () => {
  function renderForm() {
    return render(
      <FormWrapper>
        <UserEmailFormItem />
      </FormWrapper>,
    );
  }

  describe('when rendering', () => {
    it('should have the correct fields', () => {
      const { baseElement } = renderForm();
      expect(baseElement).toMatchSnapshot();

      const select = screen.getByLabelText(/User Email/i);

      expect(select).toBeInTheDocument();
    });
  });
});
