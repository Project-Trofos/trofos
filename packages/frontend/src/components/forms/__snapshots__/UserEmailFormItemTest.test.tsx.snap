// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserEmailFormItem test when rendering should have the correct fields 1`] = `
<body>
  <div>
    <form
      class="ant-form ant-form-horizontal css-dev-only-do-not-override-1qhh31y"
    >
      <div
        class="ant-form-item css-dev-only-do-not-override-1qhh31y"
      >
        <div
          class="ant-row ant-form-item-row css-dev-only-do-not-override-1qhh31y"
        >
          <div
            class="ant-col ant-form-item-label css-dev-only-do-not-override-1qhh31y"
          >
            <label
              class=""
              for="userEmail"
              title="User Email"
            >
              User Email
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control css-dev-only-do-not-override-1qhh31y"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input css-dev-only-do-not-override-1qhh31y"
                  id="userEmail"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="ant-btn css-dev-only-do-not-override-1qhh31y ant-btn-default"
        type="submit"
      >
        <span>
          submit
        </span>
      </button>
    </form>
  </div>
</body>
`;
